### ISSUE TITLE
0.7.2 - ERRO failed to send fragments to the fragment task

### ISSUE THREAD

**Issue by bjarnekvae** [input-output-hk/jormungandr#1192]
**Describe the bug**
Node crashed when using "multiple-transactions-per-slot"-script

**Mandatory Information**
1. `jcli --full-version` output; jcli 0.7.2 (HEAD-b5df0e0, release, linux [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]

2. `jormungandr --full-version` output;
jormungandr 0.7.2 (HEAD-b5df0e0, release, linux [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]

**To Reproduce**
Ran "multiple-transactions-per-slot"-script with 1000 txs. (But the previous run 100 txs went well)

**Expected behavior**
Send 1000 txs withput crash

**Additional context**
See the appended node logs + strace logs

[pushheader_failed.zip](https://github.com/input-output-hk/jormungandr/files/3886704/pushheader_failed.zip)

**Comment by mzabaluev** [comment]
The node log does not capture the panic backtrace, if there was one. Can you collect it with `RUST_BACKTRACE=1` in the environment?

**Comment by bjarnekvae** [comment]
I am running the linux-unkown-gnu binary, am I able to change the RUST_BACKTRACE variabel there?

**Comment by mzabaluev** [comment]
@bjarnekvae Yes, you can run it in the shell as `RUST_BACKTRACE=1 jormungandr ...`

**Comment by carl-IOHK** [comment]
https://iohk.zendesk.com/agent/tickets/73950

**Comment by carl-IOHK** [comment]
During heavy transaction testing on the nightly network when the fragment logs hit 118K+ fragments by Jormungandr node crashed. I am sending this request to provide the RUST_BACKTRACE in case you find it useful. Jormungandr version 0.7.0 latest release (not rc7).

Nov 20 00:49:50.347 ERRO failed to send fragments to the fragment task, reason: send failed because receiver is gone, direction: in, stream: fragments, node_id: 4c82adbedd562085a15a89c2fd20a1666a543be261cbfca6, peer_addr: 86.90.6.114:3888, task: network 
Nov 20 00:49:50.350 ERRO failed to send fragments to the fragment task, reason: send failed because receiver is gone, direction: in, stream: fragments, node_id: a9b1a04362f7171b6d7a16361209106d38918bbe0c1cd0d0, peer_addr: 185.161.193.60:8201, task: network 
Nov 20 00:49:50.353 ERRO failed to send fragments to the fragment task, reason: send failed because receiver is gone, direction: in, stream: fragments, node_id: f46e305cffa1a6481ec664500cc9dfd0a2cbffe953c08359, peer_addr: 95.216.151.45:3099, task: network 
Nov 20 00:49:50.362 ERRO failed to send fragments to the fragment task, reason: send failed because receiver is gone, direction: in, stream: fragments, node_id: 7b4325e12742546e86157ceaa72f05884aca895c23075a1c, peer_addr: 93.43.217.136:3100, task: network 
0: 0x10f2fcc05 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h66f38d4abb2e41c1 
1: 0x10f31f240 - core::fmt::write::h4f487e714088986d 
2: 0x10f2f72cb - std::io::Write::write_fmt::h7bd7b6fe8b47c5fb 
3: 0x10f2fef0a - std::panicking::default_hook::{{closure}}::heee79636c241547c 
4: 0x10f2fec15 - std::panicking::default_hook::hfcbd07059d15441e 
5: 0x10f2ff5d7 - std::panicking::rust_panic_with_hook::h0c4b67125f55410a 
6: 0x10f33c971 - std::panicking::begin_panic::h4b05cb3960ac4354 
7: 0x10ec5faa0 - jormungandr::start_services::{{closure}}::{{closure}}::h5f29b6245920c236 
8: 0x10ebaa792 - std::panicking::try::do_call::hdac22973e4a8bd4e 
9: 0x10f302e7f - __rust_maybe_catch_panic 
10: 0x10eb6ba62 - <futures::future::catch_unwind::CatchUnwind<F> as futures::future::Future>::poll::haa1078dbcd8c061a 
11: 0x10ec40eb8 - <futures::future::map_err::MapErr<A,F> as futures::future::Future>::poll::h17814ccd9fcdcf1f 
12: 0x10f2dda5d - futures::task_impl::std::set::hf6a81d7ad603e678 
13: 0x10f2dd5f6 - std::panicking::try::do_call::hcd8daf8285f047e0 
14: 0x10f302e7f - __rust_maybe_catch_panic 
15: 0x10f2dd05c - tokio_threadpool::task::Task::run::h760436a59674e20c 
16: 0x10f2d7df2 - tokio_threadpool::worker::Worker::run_task::ha7990cca7b384af0 
17: 0x10f2d7460 - tokio_threadpool::worker::Worker::run::hcce58b75f670a94e 
18: 0x10f2baf6d - std::thread::local::LocalKey<T>::with::h10e97151fce0e682 
19: 0x10f2bb233 - std::thread::local::LocalKey<T>::with::h8bd1893f4e1dcef8 
20: 0x10f2bb0e5 - std::thread::local::LocalKey<T>::with::h25cef123f17cae5c 
21: 0x10f2bcd31 - tokio::runtime::threadpool::builder::Builder::build::{{closure}}::h936dd1a535065305 
22: 0x10f2daaeb - std::thread::local::LocalKey<T>::with::h4ea373a2c47f8794 
23: 0x10f2da59b - std::thread::local::LocalKey<T>::with::h144e317d6ab1920c 
24: 0x10f2d3d70 - std::sys_common::backtrace::__rust_begin_short_backtrace::h75b0980a2f3dd595 
25: 0x10f2dd560 - std::panicking::try::do_call::h2b96ac45e1874e98 
26: 0x10f302e7f - __rust_maybe_catch_panic 
27: 0x10f2d56c7 - core::ops::function::FnOnce::call_once{{vtable.shim}}::he746ca0ef6339ffd 
28: 0x10f2f066e - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hd1afe4d791e3679f 
29: 0x10f30239e - std::sys::unix::thread::Thread:new:thread_start::h785fcf8d81bd0f52 
30: 0x7fff6b26cd36 - _pthread_start
